📝 Add reactor docs to introduction (#2265)
Fusion-Power-Plant-Framework/bluemira: documentation/source/base/components.rst

            )

A :py:class:`ComponentManager` should be how a :py:class:`Component` is used after creation within the top level of the reactor design.

Reactor
-------

:py:class:`Reactors` are again designed to be created by the :ref:`Reactor Designer <how to use>`.
This object is the complete reactor and is a container that allows easy access to any part of it.
Methods on the :py:class:`Reactor` object have access to all parts of the reactor

> enabling functionality that need to interact with multiple :py:class:`ComponentManagers`.

Reviewer: ```suggestion
enabling functionality that needs to interact with multiple :py:class:`ComponentManagers`.
```

.. code-block:: python

    class MyReactor(Reactor):
        '''An example of how to declare a reactor structure.'''

        plasma: MyPlasma
        tf_coils: MyTfCoils

        def get_ripple(self):
            '''Calculate the ripple in the TF coils.'''

    reactor = MyReactor("My Reactor")
    reactor.plasma = build_plasma()
    reactor.tf_coils = build_tf_coils()
    reactor.show_cad()

> A :py:class:`Reactor` interacts dynamically with :py:class:`ComponentManagers`.

Reviewer: Only a small thing (and may just be my personal preference), but I would usually put the `s` outside the directive, since the class name is actually `ComponentManager` (note you'll need to escape the `s` - `` :py:class:`ComponentManager`\s ``).

```suggestion
A :py:class:`Reactor` interacts dynamically with :py:class:`ComponentManager`\s.
```

If you can be bothered, you could even do a find-and-replace on these to make them `` :py:class:`~bluemira.base.reactor.ComponentManager` `` - then they'll link to the API docs.

Author: Done!

All the default methods on :py:class:`Reactor` such as :py:meth:`show_cad` will act
on the currently available :py:class:`ComponentManagers` ignoring unavailable parts
of the reactor. If a :py:class:`Component` is directly added to a :py:class:`Reactor`

> and not wrapped in a :py:class:`ComponentManagers` it will be ignored by the :py:class
> :`Reactor` methods.

Reviewer: This directive doesn't work when it's split on to two lines

```suggestion
and not wrapped in a :py:class:`ComponentManagers` it will be ignored by the
:py:class:`Reactor` methods.
```
